### ISSUE TITLE
Gem::Specification.load returning nil

### ISSUE THREAD

**Issue by JackuB** [algolia/algoliasearch-jekyll#6]
In version `0.4.2` I've encountered this error:

```
updating algolia index
/Library/Ruby/Gems/2.0.0/gems/algoliasearch-jekyll-0.4.2/lib/push.rb:129:in `set_user_agent_header': undefined method `version' for nil:NilClass (NoMethodError)
    from /Library/Ruby/Gems/2.0.0/gems/algoliasearch-jekyll-0.4.2/lib/push.rb:135:in `create_index'
    from /Library/Ruby/Gems/2.0.0/gems/algoliasearch-jekyll-0.4.2/lib/push.rb:163:in `push'
    from /Library/Ruby/Gems/2.0.0/gems/algoliasearch-jekyll-0.4.2/lib/push.rb:94:in `write'
    from /Library/Ruby/Gems/2.0.0/gems/jekyll-2.5.3/lib/jekyll/site.rb:53:in `process'
    from /Library/Ruby/Gems/2.0.0/gems/algoliasearch-jekyll-0.4.2/lib/algoliasearch-jekyll.rb:32:in `block (3 levels) in init_with_program'
    from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `call'
    from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `block in execute'
    from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `each'
    from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `execute'
    from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/program.rb:42:in `go'
    from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary.rb:19:in `program'
    from /Library/Ruby/Gems/2.0.0/gems/jekyll-2.5.3/bin/jekyll:20:in `<top (required)>'
    from /usr/bin/jekyll:23:in `load'
    from /usr/bin/jekyll:23:in `<main>'
```

Could be same issue as this one in rubygems: https://github.com/rubygems/rubygems/issues/1287

When I overwrite `version` in `push.rb` to string, it works.

**Comment by pixelastic** [comment]
Thanks for the report. Are you using bundler? 

If so, could you try with `bundle exec jekyll algolia push`?

**Comment by JackuB** [comment]
Running `bundle exec jekyll algolia push`: Fails
Running `bundle exec rake update_search`: Fails
Running `rake update_search`: Fails

This is our `Rakefile`:

``` ruby
desc 'Update algolia index'
task :update_search do
  puts 'updating algolia index'
  puts `bundle exec jekyll algolia push -t`
end
```

**Comment by pixelastic** [comment]
@JackuB Do you mind jumping on the livechat in algolia.com so I can troubleshoot that with you?

**Comment by pixelastic** [comment]
@JackuB I've just released v0.4.3 that should hopefully fix it. It no longer needs `rubygems` to known the version.

Just out of curiosity, could you send me a `ls` of the `/Library/Ruby/Gems/2.0.0/gems/algoliasearch-jekyll-0.4.2/` folder? 

I could not reproduce on my end, but I have the feeling the file I was reading is not there. Did you had the error locally on your machine or in a CI environment?

**Comment by JackuB** [comment]
Thanks! I was getting this error locally, didn't try CI - 0.4.3 works, closing this one.
